### ISSUE TITLE
are you considering releasing a standalone version that does not rely on Nerfstudio in the future?

### ISSUE THREAD

**Issue by 1397201241** [AaltoML/desplat#9]
Hello, Author! Are you considering releasing a standalone version that does not rely on Nerfstudio in the future? For example, could it be implemented primarily based on the gsplat library for its core functionality?

**Comment by johanna0626** [comment]
Hi! Thanks for your interest. Yes, we have a gsplat version with its core functionality, based on `gsplat/examples/simple_trainer.py`. There are a few differences in the training strategy (e.g., threshold and scene scale), so the results may vary slightly from those reported in the publication. We will release the code in a few weeks.

**Comment by 1397201241** [comment]
> Hi! Thanks for your interest. Yes, we have a gsplat version with its core functionality, based on `gsplat/examples/simple_trainer.py`. There are a few differences in the training strategy (e.g., threshold and scene scale), so the results may vary slightly from those reported in the publication. We will release the code in a few weeks.

Looking forward to it!

**Comment by 1397201241** [comment]
Hi! Just wanted to gently check if there's any update on this. Thanks for your time!

**Comment by johanna0626** [comment]
Hi! I have uploaded the gsplat version to this repo, please check [gsplat branch](https://github.com/AaltoML/desplat/tree/gsplat). The gsplat version is 1.0.0. There are some differences in the pipelines of newer versions, which may not be compatible with our code.

Please note that this is a reproduction, and all results in the paper were obtained using the Nerfstudio codebase. I haven’t adjusted the parameters for the gsplat version, e.g., the thresholds for splitting or cloning and the learning rate. The scene coordinates are also different from the Nerfstudio version.

I reran the experiment on the Yoda scene, and the qualitative results are shown below:

<img width="862" height="431" alt="Image" src="https://github.com/user-attachments/assets/6348d4a3-580f-4f87-ae2e-c2bf7f499f5c" />

<img width="862" height="431" alt="Image" src="https://github.com/user-attachments/assets/fade82ea-5cfc-4396-adbb-3143694a8f1e" />

These are the combined images, where the blurred parts indicate the distractors.

The core functions (e.g., splitting and cloning) can be found in `examples/simple_trainer_desplat.py`.
For dataparsers, please update `examples/datasets/colmap.py`.

Hope this helps! Feel free to reach out if you have any questions or encounter bugs.
Thank you very much for your attention!

**Comment by 1397201241** [comment]
> Hi! I have uploaded the gsplat version to this repo, please check [gsplat branch](https://github.com/AaltoML/desplat/tree/gsplat). The gsplat version is 1.0.0. There are some differences in the pipelines of newer versions, which may not be compatible with our code.
> 
> Please note that this is a reproduction, and all results in the paper were obtained using the Nerfstudio codebase. I haven’t adjusted the parameters for the gsplat version, e.g., the thresholds for splitting or cloning and the learning rate. The scene coordinates are also different from the Nerfstudio version.
> 
> I reran the experiment on the Yoda scene, and the qualitative results are shown below:
> 
> <img alt="Image" width="862" height="431" src="https://private-user-images.githubusercontent.com/79347934/501164304-6348d4a3-580f-4f87-ae2e-c2bf7f499f5c.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjA1OTkyMTEsIm5iZiI6MTc2MDU5ODkxMSwicGF0aCI6Ii83OTM0NzkzNC81MDExNjQzMDQtNjM0OGQ0YTMtNTgwZi00Zjg3LWFlMmUtYzJiZjdmNDk5ZjVjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEwMTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMDE2VDA3MTUxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTcwMzNjYzMyYjkxMWQ5MzMxY2I1YTBkNGM0Y2I3MzJhY2I4MWMwYWZlODFkMjZjMzE1Y2Y3Yzg2MjA2NjZkZTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.YyQziMH_cdnse9jSA--JsJmK5_k1NT-5DNbhjks5g1M"> <img alt="Image" width="862" height="431" src="https://private-user-images.githubusercontent.com/79347934/501164402-fade82ea-5cfc-4396-adbb-3143694a8f1e.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjA1OTkyMTEsIm5iZiI6MTc2MDU5ODkxMSwicGF0aCI6Ii83OTM0NzkzNC81MDExNjQ0MDItZmFkZTgyZWEtNWNmYy00Mzk2LWFkYmItMzE0MzY5NGE4ZjFlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEwMTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMDE2VDA3MTUxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTMzOTkwYTRjMDJlZTJiMmE3NGFjY2UwMzdlMjIyZGMxNzUyYmFkOTRhNWNjNjFlZGY5OThlN2E5NDA5ZTUwMjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.tQAjj4ZagGYdt2CKH_k4hffDC-e6mUjrfKYbFFRzGds">
> These are the combined images, where the blurred parts indicate the distractors.
> 
> The core functions (e.g., splitting and cloning) can be found in `examples/simple_trainer_desplat.py`. For dataparsers, please update `examples/datasets/colmap.py`.
> 
> Hope this helps! Feel free to reach out if you have any questions or encounter bugs. Thank you very much for your attention!


Thanks  for that and I will try it
